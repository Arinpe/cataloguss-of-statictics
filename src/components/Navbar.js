import {
  AppBar, Toolbar, makeStyles,
} from '@material-ui/core';
import { Link } from 'react-router-dom';
import React, { useEffect, useState } from 'react';
import Logo from '../assets/images/logo.png';

const useStyles = makeStyles({
  root: {
    backgroundColor: 'transparent',
    transition: 'all 1s',
    boxShadow: 'none',
    left: 0,
  },
  logo: {
    width: '140px',
    height: '30px',
  },
});

const NavBar = () => {
  const classes = useStyles();
  const [navBg, setNavBg] = useState(false);

  useEffect(() => {
    window.addEventListener('scroll', () => {
      if (window.scrollY > 50) {
        setNavBg(true);
      } else {
        setNavBg(false);
      }
    });

    return () => window.removeEventListener('scroll', () => true);
  }, []);

  return (
    <AppBar className={`${classes.root} ${navBg ? 'Nav__show' : ''} max-width`} position="fixed">
      <Toolbar variant="dense">
        <Link to="/">
          <img className={classes.logo} src={Logo} alt="Brand logo" />
        </Link>
      </Toolbar>
    </AppBar>
  );
};
export default NavBar;
