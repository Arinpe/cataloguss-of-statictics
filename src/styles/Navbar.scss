.Nav__show {
  background-color: #131313 !important;
}
