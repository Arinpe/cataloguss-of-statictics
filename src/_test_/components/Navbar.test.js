import Navbar from '../../components/Navbar';
import shallowWrapper from '../testSetup';

describe('Navigation bar', () => {
  const nav = shallowWrapper(Navbar);

  test('should render Navbar', () => {
    expect(nav).toBeTruthy();
    expect(nav).not.toBeFalsy();
    expect(nav).toMatchSnapshot();
  });

  test('should have a logo', () => {
    const logo = nav.find('img');
    expect(logo).toHaveLength(1);
    expect(logo).not.toHaveLength(0);
    expect(logo.prop('src')).toEqual('logo.png');
    expect(logo.prop('src')).not.toBeNull();
  });
});
