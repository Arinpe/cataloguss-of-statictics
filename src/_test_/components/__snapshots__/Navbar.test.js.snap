// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar should render Navbar 1`] = `ShallowWrapper {}`;
